testing.py

```diff
@@ -1426,16 +1426,31 @@ def do_post(self, url, data, *wargs, **kwargs):
         return self.responses.pop(0)
 
 
+class MockRequestsRequest(object):
+    """A mock object that simulates an HTTP request from the
+    `requests` library.
+    """
+    def __init__(self, url, method="GET", headers=None):
+        self.url = url
+        self.method = method
+        self.headers = headers or dict()
+
+
```

Comment:
This is necessary to test `_annotate_feed_response`, which looks at the Authorization header that was sent as part of the interaction with the metadata wrangler. (This, in turn, makes it easy for someone looking at a test result to duplicate the HTTP request that caused it.)